app/models/exporters/incident_recorder_exporter.rb

```diff
@@ -335,16 +335,24 @@ def props
           end,
           'service.wants_legal_action' => ->(model) do
             legal_counseling = model.try(:legal_assistance_services_subform_section)
-            if legal_counseling.present?
-              legal_actions = legal_counseling.
-                  map{|l| l.try(:pursue_legal_action)}
-              if legal_actions.include? true
-                I18n.t("exports.incident_recorder_xls.yes")
-              elsif legal_actions.include? false
-                I18n.t("exports.incident_recorder_xls.no")
-              elsif legal_actions.include? nil
-                I18n.t("exports.incident_recorder_xls.service_referral.undecided")
-              end
+            if not legal_counseling.present?
+              return ""
+            end
+
+            legal_actions = legal_counseling.
+                map{|l| l.try(:pursue_legal_action)}
+            if legal_actions.include? "true"
+              I18n.t("exports.incident_recorder_xls.yes")
+            elsif legal_actions.include? "false"
+              I18n.t("exports.incident_recorder_xls.no")
+            elsif legal_actions.include? "undecided"
+              I18n.t("exports.incident_recorder_xls.service_referral.undecided")
+            elsif legal_actions.include? nil
+              # No option selected
+              return ""
+            else
+              # Should not reach this point
+              return ""
```

Comment:
The field pursue_legal_action uses the lookup lookup-yes-no-undecided.  There is logic in the application that converts lookup-yes-no and lookup-yes-no-undecided to use boolean true/false instead of string yes/no.   This is not the correct fix.

Reply:
Thanks @relledge for point me into the right direction :-D I will update the patch.